### PR TITLE
Add openresty support

### PR DESCRIPTION
**Pull request by dubloom**
It is now possible to build a openresty-compatible version of the nginx-datadog module. 
There is no documentation in this PR as it will come with my other PR which aims at reworking the documentation.

**If you want to test locally:**

You need to create a openresty-version-info.
You need then to build a local image (as i'm not yet able to publish it) with:

RESTY_VERSION=1.25.3.2 RESTY_OPENSSL_VERSION=1.1.1w make build-openresty-toolchain
(The resty openssl version changes between targeted openresty version so you need to specify it)

You can then test with:
WAF=ON RESTY_VERSION=1.25.3.2 make openresty-test       

**When a .conf differs between openResty and NGINX:**

We create two folders a base conf for nginx and one for openresty and when we know which one will be needed we copy it in the right folder

### CODE REVIEW

File: openresty/build/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,95 @@
+FROM ubuntu:jammy
```
Comment (pablomartinezbernardo) [inline openresty/build/Dockerfile:1]: Nit: Pin it more specifically, these change

File: openresty/build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    && eval ./configure -j${RESTY_J} ${_RESTY_CONFIG_DEPS} ${RESTY_CONFIG_OPTIONS} ${RESTY_CONFIG_OPTIONS_MORE} ${RESTY_LUAJIT_OPTIONS} ${RESTY_PCRE_OPTIONS} \ 
+    && cd /tmp \
+    && rm -rf \
+        openssl-${RESTY_OPENSSL_VERSION}.tar.gz openssl-${RESTY_OPENSSL_VERSION} \
+        pcre-${RESTY_PCRE_VERSION}.tar.gz pcre-${RESTY_PCRE_VERSION} \
+        openresty-${RESTY_VERSION}.tar.gz
+
+COPY install-cmake /tmp
+RUN ["chmod", "+x", "/tmp/install-cmake"]
+RUN ["./tmp/install-cmake"]
```
Comment (pablomartinezbernardo) [inline openresty/build/Dockerfile:93]: Nit: Consistency in paths

File: Makefile [hunk 3]
```diff
@@ -118,11 +157,63 @@ build-musl-aux:
...
+		--build-arg RESTY_CONFIG_OPTIONS=${RESTY_CONFIG_OPTIONS} \
+		--build-arg RESTY_CONFIG_OPTIONS_MORE=${RESTY_CONFIG_OPTIONS_MORE} \
+		--build-arg RESTY_OPENSSL_URL_BASE=${RESTY_OPENSSL_URL_BASE} \
+		--build-arg RESTY_OPENSSL_VERSION=${RESTY_OPENSSL_VERSION} \
+		--build-arg RESTY_OPENSSL_PATCH_VERSION=${RESTY_OPENSSL_PATCH_VERSION} \
+		--build-arg RESTY_PCRE_VERSION=${RESTY_PCRE_VERSION} \
+		--build-arg RESTY_PCRE_BUILD_OPTIONS=${RESTY_PCRE_BUILD_OPTIONS} \
+		--build-arg RESTY_PCRE_SHA256=${RESTY_PCRE_SHA256} \
+		--build-arg RESTY_J=${RESTY_J} \
+		-t louis/openresty-${RESTY_VERSION} \
```
Comment (pablomartinezbernardo) [inline Makefile:181]: If you have been unable to find a way for use to publicly publish images, WDYT about building the toolchain as part of the CI and therefore doesn't need to be published?

Comment (dubloom) [reply Makefile:181]: I can publicly push images (I think at least), it is just that Damien wanted to give a look before publishing it. Building the toolchain in the CI would increase too much the duration of the CI.

File: lab/bin/run-openresty-lab [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+cd "$(dirname "$0")/../"
+
+set -e
+
+. ../openresty-version-info
+export BASE_IMAGE
+export NGINX_MODULES_PATH="${NGINX_MODULES_PATH:-/usr/local/openresty/nginx/modules/}"
+export NGINX_CONF_PATH="${NGINX_CONF_PATH:-/usr/local/openresty/nginx/conf/nginx.conf}"
+
+cp services/nginx/openresty_base.conf services/nginx/nginx.conf
```
[the remark this answers is not in the document]
Comment (dubloom) [reply lab/bin/run-openresty-lab:12]: It will not as services/nginx/nginx.conf is in gitignore

**Pull request closed without merging**